protocol-units/bridge/service/src/chains/ethereum/event_monitoring.rs

```diff
@@ -40,151 +41,241 @@ impl EthMonitoring {
+				let mut last_processed_block = 0;
+				loop {
+					let block_number = rpc_provider.get_block_number().await.unwrap();
+					if last_processed_block < block_number {
+						last_processed_block = block_number;
+						let initiator_initiate_event_filter = initiator_contract
+							.BridgeTransferInitiated_filter()
+							.from_block(BlockNumberOrTag::Number(last_processed_block));
+						// event BridgeTransferCompleted(bytes32 indexed _bridgeTransferId, bytes32 pre_image);
+						let initiator_trcompleted_event_filter = initiator_contract
+							.BridgeTransferCompleted_filter()
+							.from_block(BlockNumberOrTag::Number(last_processed_block));
+						// event BridgeTransferRefunded(bytes32 indexed _bridgeTransferId);
+						let initiator_trrefund_event_filter = initiator_contract
+							.BridgeTransferRefunded_filter()
+							.from_block(BlockNumberOrTag::Number(last_processed_block));
+						let counterpart_trlocked_event_filter = counterpart_contract
+							.BridgeTransferLocked_filter()
+							.from_block(BlockNumberOrTag::Number(last_processed_block));
+						let counterpart_trcompleted_event_filter = counterpart_contract
+							.BridgeTransferCompleted_filter()
+							.from_block(BlockNumberOrTag::Number(last_processed_block));
+						//event BridgeTransferAborted(bytes32 indexed bridgeTransferId);
+						let counterpart_trcaborted_event_filter = counterpart_contract
+							.BridgeTransferCompleted_filter()
+							.from_block(BlockNumberOrTag::Number(last_processed_block));
+
+						//Initiator event stream
+						match initiator_initiate_event_filter.query().await {
+							Ok(events) => {
+								for (initiated, _log) in events {
+									let event = {
+										// BridgeTransferInitiated(bridgeTransferId, originator, recipient, totalAmount, hashLock, initiatorTimeLockDuration);
+										let details: BridgeTransferDetails<EthAddress> =
+											BridgeTransferDetails {
+												bridge_transfer_id: BridgeTransferId(
+													*initiated._bridgeTransferId,
+												),
+												initiator_address: BridgeAddress(EthAddress(
+													Address::from(initiated._originator),
+												)),
+												recipient_address: BridgeAddress(
+													initiated._recipient.to_vec(),
+												),
+												hash_lock: HashLock(*initiated._hashLock),
+												time_lock: initiated._timeLock.into(),
+												amount: initiated.amount.into(),
+												state: 0,
+											};
+										BridgeContractEvent::Initiated(details)
+									};
+									if sender.send(Ok(event)).await.is_err() {
+										tracing::error!("Failed to send event to listener channel");
+										break;
+									}
+								}
+							}
+							Err(err) => {
+								if sender
+									.send(Err(BridgeContractError::OnChainError(err.to_string())))
+									.await
+									.is_err()
+								{
+									tracing::error!("Failed to send event to listener channel");
+									break;
```

Comment:
After pasting this config in to `bridge/config.yaml`:

```
{
  "eth": {
    "eth_rpc_connection_protocol": "https",
    "eth_rpc_connection_hostname": "ethereum-holesky-rpc.publicnode.com",
    "eth_rpc_connection_port": 443,
    "eth_ws_connection_protocol": "ws",
    "eth_ws_connection_hostname": "0.0.0.0",
    "eth_ws_connection_port": 8090,
    "eth_chain_id": 0,
    "eth_initiator_contract": "0xB7cBE82b07fdA3517AE882e12C7Faa764C16Afb9",
    "eth_counterparty_contract": "0xB5bBBd4913da0a8eb723E8cD1e02d69565f4da66",
    "eth_weth_contract": "0xB7f8BC63BbcaD18155201308C8f3540b07f84F5e",
    "signer_private_key": "0x3128c9024817f3c078695489ff3a389f0e4c27f6d364f663758fa3476564989b",
    "time_lock_secs": 60,
    "gas_limit": 10000000000000000,
    "transaction_send_retries": 10,
    "asset": "MOVE"
  },
  "movement": {
    "movement_signer_key": "0x0000000000000000000000000000000000000000000000000000000000000001",
    "movement_native_address": "0xface",
    "mvt_rpc_connection_protocol": "https",
    "mvt_rpc_connection_hostname": "testnet.bardock.movementnetwork.xyz",
    "mvt_rpc_connection_port": 443,
    "mvt_faucet_connection_protocol": "http",
    "mvt_faucet_connection_hostname": "0.0.0.0",
    "mvt_faucet_connection_port": 30732,
    "mvt_init_network": "custom",
    "rest_hostname": "127.0.0.1",
    "rest_port": 308833,
    "grpc_protocol": "http",
    "grpc_hostname": "127.0.0.1",
    "grpc_port": 50051
  },
  "testing": {
    "eth_well_known_account_private_keys": []
  }
}
```
I get the error
```
[bridge	] 2024-10-25T11:23:01.436574Z  INFO bridge_service: Bridge current transfer processing:[]
[bridge	] 2024-10-25T11:23:15.233736Z ERROR alloy_rpc_client::poller: failed to poll err=server returned an error response: error code -32000: filter not found
[bridge	] 2024-10-25T11:23:15.247208Z ERROR alloy_rpc_client::poller: failed to poll err=server returned an error response: error code -32000: filter not found
[bridge	] 2024-10-25T11:23:22.313824Z ERROR alloy_rpc_client::poller: failed to poll err=server returned an error response: error code -32000: filter not found
[bridge	] 2024-10-25T11:23:22.376266Z ERROR alloy_rpc_client::poller: failed to poll err=server returned an error response: error code -32000: filter not found
[bridge	] 2024-10-25T11:23:22.425767Z ERROR alloy_rpc_client::poller: failed to poll err=server returned an error response: error code -32000: filter not found
[bridge	] thread 'tokio-runtime-worker' panicked at /Users/andygmove/Downloads/repos/movement/protocol-units/bridge/service/src/chains/ethereum/event_monitoring.rs:132:17:
[bridge	] all branches are disabled and there is no else branch
```

where it breaks at this line, 132.

Reply:
Going to resolve this as I didn't receive this error and the tests passed, not sure what was happening locally for you. I ran the framework e2e test multiple times to be sure.